arches_resource_sets/views/resource_set_view.py

```diff
@@ -0,0 +1,66 @@
+from json import JSONDecodeError
+import logging
+import uuid
+
+from django.http import Http404
+
+from arches.app.views.api import APIBase
+from arches_resource_sets.models import ResourceSet
+from arches.app.utils.response import JSONErrorResponse, JSONResponse
+from arches.app.utils.betterJSONSerializer import JSONDeserializer, JSONSerializer
+
+logger = logging.getLogger(__name__)
+
+class ResourceSetView(APIBase):
+    def get(self, request, set_id=None):
+        if set_id is None:
+            return JSONResponse({"resource_sets": ResourceSet.objects.all()})
+        set = ResourceSet.objects.filter(id=set_id).values().first()
+        return JSONResponse({"resource_set": set})
+
+    def put(self, request, set_id):
+        try:
+            set = ResourceSet.objects.get(id=set_id)
+            request_body = JSONDeserializer().deserialize(request.body)
+            description = request_body["description"] if "description" else ""
+            set.description = description
+            set.save()
+            return JSONResponse({"resource_set": set})
+        except ResourceSet.DoesNotExist:
+            return JSONErrorResponse("Could not update resource set", "Resource set id '{}' not found".format(set_id), status=404)
+        except JSONDecodeError:
+            return JSONErrorResponse("Could not update resource set", "Invalid request body: '{}'".format(request.body), status=400) 
+        except Exception as e:
+            logger.error(e)
+            return JSONErrorResponse("Unspecified error", "Contact an Administrator", status=500)
+
+
+
+    def post(self, request):
+        set_id = uuid.uuid4()
+        set = ResourceSet.objects.create(id=set_id, owner=request.user)
+        try:
+            request_body = JSONDeserializer().deserialize(request.body)
+            description = request_body["description"] if "description" in request_body else ""
+            set.description = description
+            set.save()
+            return JSONResponse({"resource_set": set})
+        except JSONDecodeError:
+            return JSONErrorResponse("Request was malformed", "Ensure the request body is valid json", status=400)
+        except Exception as e:
+            logger.error(e)
+            return JSONErrorResponse("Unspecified error", "Contact an Administrator", status=500)
+
+
+
```

Comment:
Ok, I think this does what we want.  allow a user to skip posting a body altogether but if they do enforce that it's proper JSON.
```suggestion
    def post(self, request):
        description = ""
        set_id = uuid.uuid4()
        try:
            if request.body:
                try:
                    request_body = JSONDeserializer().deserialize(request.body)
                    description = request_body.get("description", "")
                except JSONDecodeError:
                    return JSONErrorResponse(
                        "Request was malformed",
                        "Ensure the request body is valid json",
                        status=400,
                    )
            ret = ResourceSet.objects.create(id=set_id, owner=request.user)
            ret.description = description
            ret.save()
            return JSONResponse({"resource_set": ret})
        except Exception as e:
            logger.error(e)
            return JSONErrorResponse(
                "Unspecified error", "Contact an Administrator", status=500
            )
```

Reply:
we can probably avoid the extra nesting of that inner catch and use a single jsondecodeerror in the extern - nothing else is going to throw it.

Reply:
Yep, much better!